presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarFromAnnotationsParser.java

```diff
@@ -124,7 +125,7 @@ private static SqlScalarFunction parseParametricScalar(ScalarHeaderAndMethods sc
             }
 
             signature = signature.isPresent() ? signature : Optional.of(implementation.getSignature());
-            validateSignature(signature, implementation.getSignature());
+            validateSignaturesCompatibility(signature, implementation.getSignature());
```

Comment:
Please make the same changes here as I recommended above. (I understand this is mostly pre-existing code.)